chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -869,6 +876,17 @@ impl PeerManagerActor {
                 error!(target: "network", ?err, "Failed to unban a peer");
             }
         }
+        if let Err(err) = self.peer_store.update_connected_peers_last_seen(&self.clock) {
+            error!(target: "network", ?err, "Failed to update peers last seen time.");
+        }
+
+        // Get the peers that we failed to connect to recently, and mark their connection as failed.
+        for failed_peer_id in self.failed_connections.lock().expect("Lock error").drain(..) {
+            debug!(target: "network", ?failed_peer_id, "Marking peer as failed.");
+            if self.peer_store.peer_connection_failed(&self.clock, &failed_peer_id).is_err() {
+                error!(target: "network", ?failed_peer_id, "Failed to mark peer as failed.");
+            }
+        }
```

Comment:
this should happen synchronously in unregister() call.

Reply:
or, if you want to track TCP handshake failure, synchronously below (in the place you do the push_back to the queue). There is no point in caching it.

Reply:
done.